### PR TITLE
Update repository.contents to return array for directory with single entry

### PR DESCRIPTION
**Pull request by satish-ravi**

### CODE REVIEW

**Comment by indexzero** [comment]
This is the wrong level of abstraction for this feature (which I agree with in principle). I would suggest making a PR to `unwrapper` which allows `unwrapper` to accept a predicate argument. Then this code would become:

``` js
  var cleanedPath = options.path.replace(/(^\/+)|(\/+$)/g, '');
  options.path = 'contents/'+ (options.path || '');

  return this.send(
    ['repos', project.user, project.repo, options.path],
    options,
    unwrap(args.fn, function(contents) {
      return contents[0].path === cleanedPath
    })
```

**Comment by jcrugzz** [comment]
I agree, this would make unwrapper way more useful. It currently makes too many assumptions. :+1:

**Comment by indexzero** [comment]
This also needs a test for symlink or submodule. When I add your tests to `master` they pass with the code as-is.

File: endpoints/repository.js [hunk 1]
```diff
@@ -194,12 +193,21 @@ Repository.prototype.contents = function contents(args) {
   var options = args.options = this.options(args.options, [ 'ref' ])
     , project = this.api.project(args.str);
 
+  var cleanedPath = options.path.replace(/(^\/+)|(\/+$)/g, '');
   options.path = 'contents/'+ (options.path || '');
 
   return this.send(
     ['repos', project.user, project.repo, options.path],
     options,
-    unwrap(args.fn)
```
Comment (jcrugzz) [inline endpoints/repository.js]: this should use `unwrapper@1.1.0` to handle the predicate.

**Comment by jcrugzz** [comment]
Just rebase off master to include your tests and use the new `unwrapper` with the logic you built then we'll get this merged and published!

File: test/repository.test.js [hunk 2]
```diff
@@ -73,5 +72,39 @@ describe('githulk.repository', function () {
       });
     });
 
+    it('returns object for submodule', function (next) {
+      githulk.repository.contents('satish-ravi/githulk-test', { path: 'githulk' }, function (err, result) {
```
Comment (jcrugzz) [inline test/repository.test.js]: Make a note that we should move this as a test package in the `githulks` org.

**Pull request merged**